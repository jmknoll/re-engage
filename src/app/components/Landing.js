import React, { Component } from 'react';

import {
  View,
  Text,
  StyleSheet
} from 'react-native';

import {
  LIGHT_BLUE
} from '../../shared/constants';

import Button from '../../shared/components/Button';

export default class Landing extends Component {

  constructor(props) {
    super(props)
  }

  render() {
    return(
      <View style={styles.container}>
        <Text style={styles.logo}>Re-Engage</Text>
        <Text style={styles.title}>Simplify the political engagement process in order to make your voice heard</Text>
        <View style={{width: '100%'}}>
          <Button>Get Started</Button>
          <Text style={styles.link}>Already have an account? Sign in.</Text>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    paddingLeft: 15,
    paddingRight: 15,
    flex: 1,
    backgroundColor: LIGHT_BLUE,
    justifyContent: 'space-around',
    alignItems: 'center'
  },
  logo: {
    fontSize: 40,
    color: 'white',
    textAlign: 'center',
  },
  title: {
    fontSize: 22,
    color: 'white',
    textAlign: 'center'
  },
  link: {
    marginTop: 20,
    fontSize: 16,
    color: 'white',
    textDecorationLine: 'underline',
    textAlign: 'center'
  }
})