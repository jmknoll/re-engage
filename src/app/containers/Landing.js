import { connect } from 'react-redux';

import Landing from '../components/Landing';
import * as actions from '../actions';

function mapStateToProps(state) {
  console.log('map state to props of landing')
  console.log(state)
  return {
    signedIn: state.app.signedIn,
  }
}

export default connect(mapStateToProps, actions)(Landing);