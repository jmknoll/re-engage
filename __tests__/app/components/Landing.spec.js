import 'react-native';
import React from 'react';
import Landing from '../../../src/app/components/Landing';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  return
  const tree = renderer.create(<Landing />).toJSON();
  expect(tree).toMatchSnapshot();
})