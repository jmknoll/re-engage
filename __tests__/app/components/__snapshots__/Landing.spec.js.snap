// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0097CF",
      "flex": 1,
      "justifyContent": "space-around",
      "paddingLeft": 15,
      "paddingRight": 15,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 40,
        "textAlign": "center",
      }
    }
  >
    Re-Engage
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 22,
        "textAlign": "center",
      }
    }
  >
    Simplify the political engagement process in order to make your voice heard
  </Text>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#EDF3F3",
          "borderRadius": 4,
          "height": 60,
          "justifyContent": "center",
          "marginBottom": 6,
          "marginTop": 6,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#253439",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Get Started
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "marginTop": 20,
            "textAlign": "center",
            "textDecorationLine": "underline",
          }
        }
      >
        Already have an account? Sign in.
      </Text>
    </View>
  </View>
</View>
`;
